cdap-security/src/main/java/co/cask/cdap/security/store/AbstractSecureStore.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright © 2016 Cask Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package co.cask.cdap.security.store;
+
+import co.cask.cdap.common.conf.CConfiguration;
+import co.cask.cdap.common.conf.Constants;
+
+/**
+ * Provides an abstract implementation for secure store along with some common functionality.
+ */
+public abstract class AbstractSecureStore {
+  /** Separator between the namespace name and the key name */
+  static final String NAME_SEPARATOR = ":";
+
+  private static final String KMS_BACKED = "kms";
+
+  static String getKeyName(final String namespace, final String name) {
+    return namespace + NAME_SEPARATOR + name;
+  }
+
+  public static boolean isKMSBacked(final CConfiguration cConf) {
```

Comment:
They will be inlined by the compiler any way and we may want to add more checks later. Or change the way we handle names.

> On Jul 18, 2016, at 3:48 PM, Bhooshan Mogal <notifications@github.com> wrote:
> 
> In cdap-security/src/main/java/co/cask/cdap/security/store/AbstractSecureStore.java <https://github.com/caskdata/cdap/pull/6066#discussion_r71245361>:
> 
> > +import co.cask.cdap.common.conf.Constants;
> > +
> > +/**
> > + * Provides an abstract implementation for secure store along with some common functionality.
> > + */
> > +public abstract class AbstractSecureStore {
> > +  /** Separator between the namespace name and the key name */
> > +  static final String NAME_SEPARATOR = ":";
> > +
> > +  private static final String KMS_BACKED = "kms";
> > +
> > +  static String getKeyName(final String namespace, final String name) {
> > +    return namespace + NAME_SEPARATOR + name;
> > +  }
> > +
> > +  public static boolean isKMSBacked(final CConfiguration cConf) {
> are these static methods necessary? you could just have this logic wherever you need it?
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub <https://github.com/caskdata/cdap/pull/6066/files/96308b540a6cda9ab5ee33d275c4f304922e19e0#r71245361>, or mute the thread <https://github.com/notifications/unsubscribe-auth/AAFGk1TRe8api5LLj0yBnY3NgkVI6R5Aks5qXALYgaJpZM4JOEs0>.
>